modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CategoriesBottomSheetAdapter.kt

```diff
@@ -0,0 +1,36 @@
+package au.com.shiftyjelly.pocketcasts.discover.view
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import au.com.shiftyjelly.pocketcasts.discover.R
+import au.com.shiftyjelly.pocketcasts.servers.model.DiscoverCategory
+import au.com.shiftyjelly.pocketcasts.servers.model.NetworkLoadableList
+import coil.load
+
+class CategoriesBottomSheetAdapter(val onCategoryClick: (NetworkLoadableList) -> Unit) : ListAdapter<DiscoverCategory, CategoriesBottomSheetAdapter.CategoryViewHolder>(CATEGORY_DIFF) {
+    class CategoryViewHolder(val itemView: View) : RecyclerView.ViewHolder(itemView) {
+        val categoryIcon: ImageView = itemView.findViewById(R.id.imageView)
+        val categoryName: TextView = itemView.findViewById(R.id.lblTitle)
+        val itemCategoryLayout: LinearLayout = itemView.findViewById(R.id.itemCategoryLinear)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val itemView = inflater.inflate(R.layout.item_category, parent, false)
+        return CategoryViewHolder(itemView)
+    }
+
+    override fun onBindViewHolder(holder: CategoryViewHolder, position: Int) {
+        val category = getItem(position)
+        holder.categoryName.text = category.name
+        holder.categoryName.contentDescription = category.name
+        holder.categoryIcon.load(category.icon)
+        holder.itemCategoryLayout.setOnClickListener { onCategoryClick(category) }
```

Comment:
Setting listeners during binding is not a good pattern. It will create a new object for listener every time a view is being bound. Instead it should be handled during creation in a pattern similar to this one.

```kotlin
class Holder(
    view: View,
    onClickItem: (Item) -> Unit,
) : RecyclerView.ViewHolder(view) {
    private var item: Item?
    
    init {
        viewToClick.setOnClickListener { 
            item?.let(onClickItem)
        }
    }
    
    fun bind(item: Item) {
        this.item = item
    }
}
```